system/ee/legacy/libraries/Functions.php

```diff
@@ -471,33 +502,79 @@ public function form_declaration($data)
 
-
-		if ($data['action'] == '')
+        
+		if ($data['enctype'] == 'multi' OR strtolower($data['enctype']) == 'multipart/form-data')
 		{
-			$data['action'] = $this->fetch_site_index();
+            $data['enctype'] = ' enctype="multipart/form-data" ';
 		}
-
-		if ($data['onsubmit'] != '')
+        
+		if ($data['action'] == '')
 		{
-			$data['onsubmit'] = 'onsubmit="'.trim($data['onsubmit']).'"';
+            $data['action'] = $this->fetch_site_index();
 		}
-
+        
 		if (substr($data['action'], -1) == '?')
 		{
-			$data['action'] = substr($data['action'], 0, -1);
+            $data['action'] = substr($data['action'], 0, -1);
 		}
-
-		$data['name']	= (isset($data['name']) && $data['name'] != '') ? 'name="'.$data['name'].'" '	: '';
-		$data['id']		= ($data['id'] != '') 							? 'id="'.$data['id'].'" ' 		: '';
-		$data['class']	= ($data['class'] != '')						? 'class="'.$data['class'].'" '	: '';
-
-		if ($data['enctype'] == 'multi' OR strtolower($data['enctype']) == 'multipart/form-data')
+        
+		if ($data['onsubmit'] != '')
 		{
-			$data['enctype'] = 'enctype="multipart/form-data" ';
-		}
-
-		$form  = '<form '.$data['id'].$data['class'].$data['name'].'method="post" action="'.$data['action'].'" '.$data['onsubmit'].' '.$data['enctype'].">\n";
+            $data['onsubmit'] = ' onsubmit="'.trim($data['onsubmit']).'"';
+        }
+        
+        // Next section is for the 'pass-through' functionality
+
+        $_tagparams = array();
+        $_pass_thru = '';
+        $valid_html_attributes = array();
+                
+        // $valid_html_attributes is an array of valid HTML attributes that might be specified in a FORM tag (this list has all $deft values removed, and does not include
+        // data-* or aria-* as these are processed differently).  The list taken from https://developer.mozilla.org/en-US/docs/Web/HTML/Element/form on 5/june/20
+
+        $valid_html_attributes = explode('|','accept|accept-charset|accesskey|autocapitalize|autocomplete|contenteditable|contextmenu|dir|draggable|dropzone|exportparts|hidden|inputmode|is|itemid|itemprop|itemref|itemscope|itemtype|lang|method|novalidate|onabort|onautocomplete|onautocompleteerror|onblur|oncancel|oncanplay|oncanplaythrough|onchange|onclick|onclose|oncontextmenu|oncuechange|ondblclick|ondrag|ondragend|ondragenter|ondragexit|ondragleave|ondragover|ondragstart|ondrop|ondurationchange|onemptied|onended|onerror|onfocus|oninput|oninvalid|onkeydown|onkeypress|onkeyup|onload|onloadeddata|onloadedmetadata|onloadstart|onmousedown|onmouseenter|onmouseleave|onmousemove|onmouseout|onmouseover|onmouseup|onmousewheel|onpause|onplay|onplaying|onprogress|onratechange|onreset|onresize|onscroll|onseeked|onseeking|onselect|onshow|onsort|onstalled|onsubmit|onsuspend|ontimeupdate|ontoggle|onvolumechange|onwaiting|part|rel|slot|spellcheck|style|tabindex|target|title|translate|role');
```

Comment:
This means EE is maintaining a list of valid values, and a user can't pass any custom values through. Even if EE went with this approach, this list should be an array in a config file somewhere, not a scalar value buried deep within EE's core.

Reply:
Have pushed an update that moves the array to a config file.